vllm/worker/worker.py

```diff
@@ -288,11 +295,11 @@ def execute_model(
             return []
 
         output = self.model_runner.execute_model(seq_group_metadata_list,
-                                                 self.gpu_cache)
+                                                 self.gpu_cache, num_steps)
 
-        # Worker only supports single-step execution. Wrap the output in a list
-        # to conform to interface.
-        return [output]
+        # For worker only supports single-step execution, Wrap the output
+        # in a list to conform to interface.
+        return output if isinstance(output, list) else [output]
```

Comment:
Seems the model runner should return List, then the Worker returns the model runner output.